server/src/main/scala/actors/queues/QueueLikeActor.scala

```diff
@@ -25,11 +27,24 @@ object QueueLikeActor {
 
   case object ReadyToEmit
 
-  case class UpdatedMillis(millis: Iterable[MillisSinceEpoch])
+  trait UpdateAffect {
+    def ++(other: UpdateAffect): UpdateAffect
+  }
+
+  object UpdatedMillis {
+    val empty: UpdatedMillis = UpdatedMillis(Seq())
+  }
+
+  case class UpdatedMillis(affects: Iterable[MillisSinceEpoch]) extends UpdateAffect {
```

Comment:
Are these `UpdatedMillis` or are they `DaysToReCrunch`?

Reply:
They're updated millis, ie all milliseconds that have been updated.

The way we think about this depends on where we want the responsibility to lie for deciding when a day needs to be re-crunched. At the moment that lies with the CrunchQueueActor which is informed of any milliseconds that have been updated and maintains a list of CrunchRequests

Reply:
ok cool, that makes sense.